pkg/clients/dynatrace/dynatrace_client_test.go

```diff
@@ -267,3 +281,121 @@ func createTestDynatraceClientWithFunc(t *testing.T, handler http.HandlerFunc) (
 
 	return dynatraceServer, dynatraceClient
 }
+
+func testServerErrors(t *testing.T) {
```

Comment:
these tests feel a bit overcomplicated, you are basically testing the fact that `json.Unmarshal(response, &se)` works for a `struct` that you created via `json.Marshal(&message)`, with a mocked webserver in between 😅 

Valid tests for this IMO would be:
1. Have a valid `json` object from the API (with the error you want), and try to `json.Unmarshal` that, to check that your `struct` matches the `json` object
2.  Unittests for the method `func (e ServerError) Error() string `,

Reply:
right, raw strings are returned by mocked server